src/common/snippets/src/pass/matmul_transpose.cpp

```diff
@@ -0,0 +1,71 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <snippets/itt.hpp>
+#include "snippets/remarks.hpp"
+
+#include "snippets/pass/matmul_transpose.hpp"
+#include "snippets/op/subgraph.hpp"
+#include "snippets/snippets_isa.hpp"
+#include "snippets/utils.hpp"
+
+#include <ngraph/opsets/opset8.hpp>
+#include <ngraph/rt_info.hpp>
+#include <ngraph/pattern/op/wrap_type.hpp>
+#include <ngraph/pattern/op/or.hpp>
+#include <ngraph/validation_util.hpp>
+
+
+
+ngraph::snippets::pass::MatMulTranspose::MatMulTranspose() {
+    MATCHER_SCOPE(MatMulTranspose);
+
+    auto m_matmul0 = std::make_shared<ngraph::opset1::MatMul>(
+            ngraph::pattern::any_input(ngraph::pattern::has_static_shape()),
+            ngraph::pattern::any_input(ngraph::pattern::has_static_shape()));
+
+    register_matcher(std::make_shared<ngraph::pattern::Matcher>(m_matmul0, matcher_name),
+        [=](ngraph::pattern::Matcher &m) {
+        OV_ITT_SCOPED_TASK(ngraph::pass::itt::domains::SnippetsTransform, "Snippets::op::MatMulTranspose")
+        auto root = m.get_match_root();
+
+        auto matmul0 = ngraph::as_type_ptr<ngraph::opset1::MatMul>(root);
+        if (!matmul0 || !matmul0->get_transpose_b())
+            return false;
+        auto parent1 = matmul0->get_input_node_shared_ptr(1);
+        auto transpose1 = ngraph::as_type_ptr<ngraph::opset1::Transpose>(parent1);
+        while (!transpose1 && !ov::is_type<ngraph::opset1::Parameter>(parent1)) {
+            // We can set supported order and transposed_b(false) only if ops have scalar shapes to avoid shape mismatching
+            const auto parent_count = parent1->inputs().size();
+            bool are_weights_scalar = true;
+            if (parent_count > 1) {
+                for (size_t i = 1; i < parent_count; ++i) {
+                    are_weights_scalar = are_weights_scalar && ngraph::shape_size(parent1->get_input_shape(i)) == 1;
+                }
+            }
+            if (!are_weights_scalar)
+                break;
+
+            parent1 = parent1->get_input_node_shared_ptr(0);
+            transpose1 = ngraph::as_type_ptr<ngraph::opset1::Transpose>(parent1);
+        }
+        if (!transpose1)
+            return false;
+
+        const auto transpose_pattern = ngraph::as_type_ptr<ngraph::opset1::Constant>(transpose1->get_input_node_shared_ptr(1));
+        // TODO: Reuse supported order from Transpose decomposition
```

Comment:
As suggested above, this should be in a separate transpose-fusing pass. Moreover, I'm not sure that this pass should check if we can decompose a particular `Transpose` or not - that's the tokenizer job (or, if smth goes wrong, we'll get a error on the backend side)

Reply:
As we discussed, I have created the ticket 100448 for support of MatMul with transposed inputs